src/spyglass/spikesorting/merge.py

```diff
@@ -56,10 +59,63 @@ def get_recording(cls, key):
         query = source_table & cls.merge_get_part(key)
         return query.get_recording(query.fetch("KEY"))
 
+    @classmethod
     def get_sorting(cls, key):
         """get the sorting associated with a spike sorting output"""
         source_table = source_class_dict[
             to_camel_case(cls.merge_get_parent(key).table_name)
         ]
         query = source_table & cls.merge_get_part(key)
         return query.get_sorting(query.fetch("KEY"))
+
+    @classmethod
+    def get_spike_times(cls, key):
+        spike_times = []
+        for nwb_file in cls.fetch_nwb(key):
+            if "object_id" in nwb_file:
+                # v1 spikesorting
+                spike_times.extend(
+                    nwb_file["object_id"]["spike_times"].to_list()
+                )
+            elif "units" in nwb_file:
+                # v0 spikesorting
+                spike_times.extend(nwb_file["units"]["spike_times"].to_list())
+        return spike_times
+
+    @classmethod
+    def get_spike_indicator(cls, key, time):
+        time = np.asarray(time)
+        min_time, max_time = time[[0, -1]]
+        spike_times = cls.get_spike_times(key)
+        spike_indicator = np.zeros((len(time), len(spike_times)))
+
+        for ind, times in enumerate(spike_times):
+            times = times[
+                np.logical_and(spike_times >= min_time, spike_times <= max_time)
+            ]
+            spike_indicator[:, ind] = np.bincount(
+                np.digitize(times, time[1:-1]),
+                minlength=time.shape[0],
+            )
+
+        return spike_indicator
+
+    @classmethod
+    def get_firing_rate(cls, key, time, multiunit=False):
+        spike_indicator = cls.get_spike_indicator(key, time)
+        if spike_indicator.ndim == 1:
+            spike_indicator = spike_indicator[:, np.newaxis]
+
+        sampling_frequency = 1 / np.median(np.diff(time))
```

Comment:
I ran into an error in lfp where sampling freq needed to be forced as int. Plz ignore if not true in this case